snmpman/src/main/java/com/oneandone/snmpman/snmp/MOGroup.java

```diff
@@ -144,8 +149,8 @@ public boolean next(final SubRequest request) {
     @Override
     public void prepare(SubRequest request) {
         RequestStatus status = request.getStatus();
-        if (request.getIndex() > 0) {
-            //Skip rowStatusColumn SubRequest with index 0
+        if (isSimpleSetOperation(request) || request.getIndex() > 0) {
```

Comment:
You're replacing every `(request.getIndex() > 0)` condition with a `(isSimpleSetOperation(request) || request.getIndex() > 0)` condition.

It would be shorter and easier to put all logic of the condition inside the `isSimpleSetOperation()` method and give it a good name.